### DISCUSSION TITLE
Plot Modbus Inputs

### DISCUSSION THREAD

**Discussion by zlmink** [artisan-roaster-scope/artisan#615] [Q&A]
How can I plot Modbus TCP input registers?

**Comment by poundy** [comment]
sorry, can you explain what you want to do? Do you have a different register to "normal" for (for example) bean temp , that you want to plot? Or you have a register value for a different variable that your platform supports (eg ambient air pressure) and you'd like to display that?

**Reply by zlmink** [reply]
Hi @poundy, exactly, I have a few different variables that are supported by the platform and I'd like to display them.

**Reply by poundy** [reply]
sorry, so the second? Give us a concrete example of what sensor you're talking about. 

You may need to keep in mind that Artisan is not intended to be a general datalogging/display application so some data types may not make sense in a roasting context...

**Comment by fcebreros** [comment]
Hi,
you can plot anything adding an extra device in Config/Extra Devices
[image: image.png]
There I'm plotting inputs 3 and 4 configured in Ports Configuration Modbus
tab
[image: image.png]
Hope that helps
Greetings
Frank

**Reply by poundy** [reply]
Just a note Frank, I suspect you replied in email, and your inline images never make it into the github reply. I suspect they would be VERY handy for our original poster, so any chance you could pop over here and add them by hand?

**Reply by fcebreros** [reply]
![image(6)](https://user-images.githubusercontent.com/56903027/120210839-dafb9080-c1f5-11eb-88e5-ca34b00051db.png)
![image(7)](https://user-images.githubusercontent.com/56903027/120210849-ddf68100-c1f5-11eb-806d-ac3c865dac33.png)

**Reply by poundy** [reply]
Perfect !
@zlmink  this should show you what is possible. Does that match your need?

**Comment by fcebreros** [comment]
Ok, I have attached the images here.
Does this work?

**Reply by poundy** [reply]
no, it seems like they did not. From the text in the reply i can see it still kind of looks like an email reply.  If you go to the github issue at https://github.com/artisan-roaster-scope/artisan/discussions/615 and then at the bottom you can insert images in the reply boxes there (through drag and drop or copy/paste).

**Reply by zlmink** [reply]
Hi @poundy and @fcebreros , thanks for this.  It looks like exactly what I'm looking for.  For some reason however, I'm using Modbus TCP and the data isn't showing up when I start the roast.  I have confirmed that the Modbus registers contain data via the Modbus scan feature and I have scaled them such that they should show up in the plot area, but the plot is blank.  I have also used the same Device configuration as mentioned.  Is there some way to debug the Modbus TCP 'linkage' to the plot?  Are there any other requirements for getting data to show up on the plot?  Right now, I'm just testing out a Modbus TCP server I created for collecting temperature data.

**Comment by fcebreros** [comment]
Hi,
would need more information to help you with the modbus configuration.
Please note that the configuration I have on the images is specific to my
hardware and won't work with yours.
You should check first if you see any data on the LCD's to confirm your
ports are correctly setup to work when your hardware.
Best
Frank

**Comment by nguyencongsinh** [comment]
Hi. please tell me how I can drag sliders on Artisan software to control PLC S7 1200? What data does it need to read?
Thank you!

**Reply by poundy** [reply]
You should start your own thread. This one is unrelated to your issue. Be sure to clearly define what roaster you have and how it is supporting Artisan already. If you're representing a roaster manufacturer here, you can engage the Artisan team to add support for your roaster in the project. If you're just trying to do this on your own, then there are many previous posts you can locate here that talk about how other people have investigated enabling Artisan to communicate with the PID
